### ISSUE TITLE
Feature : Select multiple "learn" by Quill content

### ISSUE THREAD

**Issue by goninja808** [lakesare/memcode#171]
AS a user of Memcode, WHEN "/learn" screen present me a list of elements of a course, I should be allowed to search TEXT present in some of them in order to SELECT them for learning / unlearning.
Simplest minimum design could be to have simple "search" field (same used to search course) to select in "/learn" specific tag / text ... "#xxx#", "xxx". Same search could allow to select/learn/unlearn all elements if submitted empty.
For instance, would be useful to select a kind of knowledge "#core knowledge#", a level of complexity "#easy#", or a step in training "#newcomer#".

**Comment by goninja808** [comment]
# first example FLASHCARDS TO LEARN as default
  Scenario: User access list of quill of course ("/learn")
    When the user fills a search field and submit it via
     enter (or side button) correspond to "learn" individual quills previously in unlearn or ignore or learn states
    Then the quill with corresponding content in the list are marked as "learn" 
    
  # second example LEARNED FLASHCARDS as optional
  Scenario: User access list of quill of course ("/learn")
    When the user fills a search field and submit it via
     enter (or side button) correspond to "unlearn" individual quills previously in unlearn or ignore or learn states
    Then the quill with corresponding content in the list are marked as "unlearn"  

  # second example IGNORED FLASHCARDS as optional
  Scenario: User access list of quill of course ("/learn")
    When the user fills a search field and submit it
     enter (or side button) correspond to "ignore" individual quills previously in learn or unlearn or ignore states
    Then the quill with corresponding content in the list are marked as "unlearn"

**Comment by lakesare** [comment]
Hm, so you'd like bulk "learning" based on a search query match.
It's possible to implement this functionality as you suggest it, but the more long-term solution is connected to the following 2 pieces of functionality:
- bulk learning (it's been requested multiple times, people clearly feel the need for it, and it's easy to implement)
- groups of flashcards within the course (this has been requested many times too, e.g. see https://github.com/lakesare/memcode/issues/5)

Before embarking upon these tasks, however, I would implement *groups of courses*, because there is a chance that *groups of flashcards within the course* won't be as needed if our courses have some more structure to them.

**Comment by goninja808** [comment]
Thank you for your prompt reply, Evgenia.
I would say that the "group of courses" functionality is certainly the best
idea for very organized people around a known / defined syllabus. You know
your users better, it is perfect if they use the application as content
providers wanting to organize in a pre-structured way.

I am more than happy if you think that it can be a priority to make quill
selection a bit more flexible : My idea of "tags" or "text" based search at
that precise level was kind of  "quick win" proposal for any user wanting
to learn "all at once" (select empty) or to "divide to conquer" with tags
representing steps, difficulties or sections.

If I wanted to participate and propose an implementation of such
solution would it be ok to have your help and how would I proceed
for the benefit of your team work ? Making a fork to modify only previously
segregated component structure would be ok ? Do you have coding style rules
or preference : would it be with some specific pattern or design to follow?

Best Regards,
Jason Gonin

Le mar. 18 oct. 2022 à 18:13, Evgenia Karunus ***@***.***> a
écrit :

**Comment by lakesare** [comment]
> If I wanted to participate and propose an implementation of such solution would it be ok to have your help and how would I proceed for the benefit of your team work ?

Yep, absolutely! We can have a call discussing the functionality/what kind of work you'd like to do.
In general ask any questions in Slack, I'll be quicker to respond there.

> Making a fork to modify only previously segregated component structure would be ok ?

Yep! Making a fork to modify anything is ok, just discuss the functionality with me if it's something major/not an obvious bug.

> Do you have coding style rules or preference : would it be with some specific pattern or design to follow?

We have eslint, and we follow a particular folder structure - you should be good if you just imitate the existing structure.

**Comment by lakesare** [comment]
### Price assignment (re: [memcode#requesting-a-feature](https://github.com/lakesare/memcode#requesting-a-feature))

**Search on the /learn screen + bulk learning all found flashcards + bulk ignoring all found flashcards**: **$150**
